sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1468 @@
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+== Contact
+
+Jessica Davies, Intel (jessica 'dot' davies 'at' intel 'dot' com)
+
+== Contributors
+
+Jessica Davies, Intel +
+Joe Garvey, Intel +
+Michael Kinsner, Intel +
+Aditi Kumaraswamy, Intel +
+Steffen Larsen, Intel +
+Gregory Lueck, Intel +
+John Pennycook, Intel +
+Roland Schulz, Intel +
+Jason Sewall, Intel +
+Abhishek Tiwari, Intel +
+Sherry Yuan, Intel
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, revision 5.
+All references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+- link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties]
+- link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr]
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Overview
+
+This extension introduces USM memory allocation functions with support for compile-time-constant and runtime properties, as defined in the link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties] extension.
+The USM memory allocation functions introduced by this extension take a `properties` object as a parameter, and return an `annotated_ptr`.
+This allows both runtime and compile-time-constant properties to be specified when allocating USM memory.
+Furthermore, the `annotated_ptr` returned by the USM memory allocation functions allows compile-time-constant information to propagate to the device compiler and thereby enable additional optimization of kernel code.
+This extension also introduces a new compile-time constant property `sycl::ext::oneapi::experimental::usm_kind`, whose single parameter is a value from the enumeration `sycl::usm::alloc`.
+
+[NOTE]
+====
+In this document, we use the shortened form `annotated_ptr` to refer to the
+proposed `sycl::ext::oneapi::experimental::annotated_ptr` class.
+We also use the shortened form `properties` to refer to the `sycl::ext::oneapi::experimental::properties` class.
+====
+
+The purpose of this document is to clearly describe and specify USM memory allocation functions with `properties` support,
+and related concepts, types, and mechanisms, and to give examples and context for their usage.
+
+== Examples
+
+Runtime and compile-time constant properties can be passed to the USM memory allocation functions introduced by this extension.
+Properties passed to an allocation function may or may not appear on the returned `annotated_ptr` object:
```

Comment:
```suggestion
Properties passed to an allocation function may or may not appear on the returned `annotated_ptr` object;
```

Reply:
Why isn't a colon appropriate here?

Reply:
The part before the punctuation doesn't introduce the part after.  Each part is a full sentence.  You could use a period, but a semi-colon seems more appropriate since you want to show that they are related.

Reply:
I changed my mind after looking at more style guides.  I think the colon is fine here.